### ISSUE TITLE
JS ERROR: TypeError: this._workspacesViews[i] is undefined

### ISSUE THREAD

**Issue by brianjmurrell** [mzur/gnome-shell-wsmatrix#55]
After updating to the latest I am getting:

```
Sep 18 09:04:14 pc gnome-shell[18671]: GNOME Shell started at Wed Sep 18 2019 09:04:06 GMT-0400 (EDT)
Sep 18 09:04:18 pc gnome-shell[18671]: JS WARNING: [resource:///org/gnome/shell/ui/workspacesView.js 731]: reference to undefined property 1
Sep 18 09:04:18 pc gnome-shell[18671]: JS ERROR: TypeError: this._workspacesViews[i] is undefined
                                                       _updateWorkspacesActualGeometry@resource:///org/gnome/shell/ui/workspacesView.js:731:13
                                                       vfunc_allocate@resource:///org/gnome/shell/ui/overviewControls.js:388:9
                                                       _updateWorkspacesViews@resource:///org/gnome/shell/ui/workspacesView.js:637:13
                                                       show@resource:///org/gnome/shell/ui/workspacesView.js:557:9
                                                       show@resource:///org/gnome/shell/ui/viewSelector.js:268:9
                                                       _animateVisible@resource:///org/gnome/shell/ui/overview.js:527:9
                                                       show@resource:///org/gnome/shell/ui/overview.js:513:9
                                                       toggle@resource:///org/gnome/shell/ui/overview.js:636:13
                                                       _initializeUI/<@resource:///org/gnome/shell/ui/main.js:190:13
Sep 18 09:04:18 pc gnome-shell[18671]: JS WARNING: [resource:///org/gnome/shell/ui/workspacesView.js 714]: reference to undefined property 1
Sep 18 09:04:18 pc gnome-shell[18671]: JS ERROR: TypeError: this._workspacesViews[i] is undefined
                                                       _updateWorkspacesFullGeometry@resource:///org/gnome/shell/ui/workspacesView.js:714:13
                                                       setWorkspacesFullGeometry@resource:///org/gnome/shell/ui/workspacesView.js:704:9
                                                       setWorkspacesFullGeometry@resource:///org/gnome/shell/ui/viewSelector.js:293:9
                                                       _updateWorkspacesGeometry@resource:///org/gnome/shell/ui/overviewControls.js:456:9
                                                       vfunc_allocate@resource:///org/gnome/shell/ui/overviewControls.js:389:9
                                                       _updateWorkspacesViews@resource:///org/gnome/shell/ui/workspacesView.js:637:13
                                                       show@resource:///org/gnome/shell/ui/workspacesView.js:557:9
                                                       show@resource:///org/gnome/shell/ui/viewSelector.js:268:9
                                                       _animateVisible@resource:///org/gnome/shell/ui/overview.js:527:9
                                                       show@resource:///org/gnome/shell/ui/overview.js:513:9
                                                       toggle@resource:///org/gnome/shell/ui/overview.js:636:13
                                                       _initializeUI/<@resource:///org/gnome/shell/ui/main.js:190:13
Sep 18 09:04:19 pc gnome-shell[18671]: Could not load a pixbuf from icon theme.
                                                       This may indicate that pixbuf loaders or the mime database could not be found.
```

Even after an `ALT-F2+r` restart.

**Comment by mzur** [comment]
1. Do you have the workspaces grid in the activity overview enabled?
2. Do the errors only appear if you have this enabled?
3. Do you have multiple monitors?
4. Does the error happen on a specific action?

**Comment by brianjmurrell** [comment]
> 1. Do you have the workspaces grid in the activity overview enabled?

Yes.  But I don't think I had it enabled when I originally reported this bug.

> 2. Do the errors only appear if you have this enabled?

No.  Enabled I get this one error when I activate the overview:

```
Sep 18 14:52:25 pc gnome-shell[18671]: JS ERROR: TypeError: this._workspacesViews[i] is undefined
                                                       _updateWorkspacesFullGeometry@resource:///org/gnome/shell/ui/workspacesView.js:714:13
                                                       setWorkspacesFullGeometry@resource:///org/gnome/shell/ui/workspacesView.js:704:9
                                                       setWorkspacesFullGeometry@resource:///org/gnome/shell/ui/viewSelector.js:293:9
                                                       _updateWorkspacesGeometry@resource:///org/gnome/shell/ui/overviewControls.js:456:9
                                                       vfunc_allocate@resource:///org/gnome/shell/ui/overviewControls.js:389:9
                                                       _updateWorkspacesViews@resource:///org/gnome/shell/ui/workspacesView.js:637:13
                                                       show@resource:///org/gnome/shell/ui/workspacesView.js:557:9
                                                       show@resource:///org/gnome/shell/ui/viewSelector.js:268:9
                                                       _animateVisible@resource:///org/gnome/shell/ui/overview.js:527:9
                                                       show@resource:///org/gnome/shell/ui/overview.js:513:9
                                                       toggle@resource:///org/gnome/shell/ui/overview.js:636:13
                                                       _initializeUI/<@resource:///org/gnome/shell/ui/main.js:190:13
```

Disabled I get two errors on activating the overview:

```
Sep 18 14:52:36 pc gnome-shell[18671]: JS ERROR: TypeError: this._workspacesViews[i] is undefined
                                                       _updateWorkspacesActualGeometry@resource:///org/gnome/shell/ui/workspacesView.js:731:13
                                                       vfunc_allocate@resource:///org/gnome/shell/ui/overviewControls.js:388:9
                                                       _updateWorkspacesViews@resource:///org/gnome/shell/ui/workspacesView.js:637:13
                                                       show@resource:///org/gnome/shell/ui/workspacesView.js:557:9
                                                       show@resource:///org/gnome/shell/ui/viewSelector.js:268:9
                                                       _animateVisible@resource:///org/gnome/shell/ui/overview.js:527:9
                                                       show@resource:///org/gnome/shell/ui/overview.js:513:9
                                                       toggle@resource:///org/gnome/shell/ui/overview.js:636:13
                                                       _initializeUI/<@resource:///org/gnome/shell/ui/main.js:190:13
Sep 18 14:52:36 pc gnome-shell[18671]: JS ERROR: TypeError: this._workspacesViews[i] is undefined
                                                       _updateWorkspacesFullGeometry@resource:///org/gnome/shell/ui/workspacesView.js:714:13
                                                       setWorkspacesFullGeometry@resource:///org/gnome/shell/ui/workspacesView.js:704:9
                                                       setWorkspacesFullGeometry@resource:///org/gnome/shell/ui/viewSelector.js:293:9
                                                       _updateWorkspacesGeometry@resource:///org/gnome/shell/ui/overviewControls.js:456:9
                                                       vfunc_allocate@resource:///org/gnome/shell/ui/overviewControls.js:389:9
                                                       _updateWorkspacesViews@resource:///org/gnome/shell/ui/workspacesView.js:637:13
                                                       show@resource:///org/gnome/shell/ui/workspacesView.js:557:9
                                                       show@resource:///org/gnome/shell/ui/viewSelector.js:268:9
                                                       _animateVisible@resource:///org/gnome/shell/ui/overview.js:527:9
                                                       show@resource:///org/gnome/shell/ui/overview.js:513:9
                                                       toggle@resource:///org/gnome/shell/ui/overview.js:636:13
                                                       _initializeUI/<@resource:///org/gnome/shell/ui/main.js:190:13

```

> 3. Do you have multiple monitors?

Yes.  Two.

> 4. Does the error happen on a specific action?

Only seeing it on overview activation and shell startup so far.

As an aside, what is the functionality when "Show workspace grid in the overview" is disabled?

**Comment by ebeem** [comment]
I can see this, always thought it was a GNOME issue and not related to this extension, because I get it when the extension is enabled or disabled as far as I remember.

**Comment by mzur** [comment]
If the error occurs when the extension is disabled, it's an upstream GNOME issue. @brianjmurrell Can you confirm this?

**Comment by ebeem** [comment]
I have just checked again, and I confirm that I get similar results even with all extensions disabled and shell restarted.
I am running `Arch Linux` `GNOME Shell 3.32.2`
my virtual machine `Ubuntu 19.04` `GNOME Shell 3.32.0` doesn't produce any errors

**Comment by mzur** [comment]
Thanks @ebeem. I also found these related issues [[1](https://gitlab.gnome.org/GNOME/gnome-shell/issues/1065), [2](https://gitlab.gnome.org/GNOME/gnome-shell/issues/1304)]. @brianjmurrell It's probably best if you report these errors there. Maybe this will be fixed in GNOME 3.34.

**Comment by ebeem** [comment]
just for future reference, I am not seeing this anymore on `GNOME 3.34`
